ruby/lib/minitest/queue/build_status_reporter.rb

```diff
@@ -25,7 +25,7 @@ def report
       end
 
       def success?
-        errors == 0 && failures == 0
+        build.error_reports.empty?
```

Comment:
The most important change is here. All the rest is mostly for display / stats purpose.